conans/client/tools/scm.py

```diff
@@ -146,4 +157,114 @@ def get_branch(self):
             branch = status.splitlines()[0].split("...")[0].strip("#").strip()
             return branch
         except Exception as e:
-            raise ConanException("Unable to get git branch from %s\n%s" % (self.folder, str(e)))
+            raise ConanException("Unable to get git branch from %s: %s" % (self.folder, str(e)))
+
+    def _check_git_repo(self):
+        try:
+            self.run("status")
+        except Exception:
+            raise ConanException("Not a valid git repository")
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+    file_protocol = 'file:///' if platform.system() == "Windows" else 'file://'
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache".format(command))
+
+    def checkout(self, url, revision="HEAD"):
+        output = ""
+        try:
+            self._check_svn_repo()
+        except ConanException:
+            params = " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
```

Comment:
I get the follwing error message (Win10, Subversion 1.9.7, Python 3.7, only using Conan.exe, not at debugging):

> svn: E205000: --trust-server-cert-failures requires --non-interactive

When changing it to
`params = " --trust-server-cert-failures=unknown-ca --non-interactive" if not self._verify_ssl else ""`
it works.

Reply:
After removing the ``--non-interactive`` argument, the behavior of ``co`` is different, this is what the help says:

```
--trust-server-cert-failures ARG : with --non-interactive, accept SSL server
                             certificates with failures; ARG is comma-separated
                             list of 'unknown-ca' (Unknown Authority),
                             'cn-mismatch' (Hostname mismatch), 'expired'
                             (Expired certificate), 'not-yet-valid' (Not yet
                             valid certificate) and 'other' (all other not
                             separately classified certificate errors).
```

I wonder if we should enumerate justt ``unknown-ca`` or someone else.

Reply:
I tested this with a local repository `file:///....`. May be we should exclude this from the `--trust-server-cert-failures`. Or better, just add it for "https"?

Reply:
I don't feel comfortable having more logic than needed, you never know when to stop adding `ifs`. 

What about doing always the `--non-interactive` AND the `--trust-server-cert-failures=unknown-ca` (add them also to the `update` and `revert`)?  What can go wrong? It will fail for a user not authenticated in the SVN repository (and maybe in a repo with a conflicted state), but then it's up to the user to solve them: save credentials and solve conflicts before running again the conan command.

Reply:
Go for it. Anyway the user can activate `verify_ssl`.
I don't think `update` and `revert` support this.

Reply:
@jgsogo 
The `--trust-server-cert-failures` is an SVN feature since 1.9.x.
We have some "older" build slaves (with Ubuntu 14.04 and 12.04, because auf gcc 4.x and compatibility issues [sorry I can't change it :cry:]) which have SVN 1.8 and lower installed. Because `--trust-server-cert` is still available in the latest SVN version (just marked as deprecated) it would make sense to use this instead of `--trust-server-cert-failures`, isn't it? Otherwise this feature will only work on newer clients.
The only workaround would be to detect SVN version first and selecting the best match.

Reply:
We will never finish this feature, 😆 

Yes, I think it makes sense to use the deprecated one... until it disappears from newer versions. It makes sense to identify the SVN version.